### ISSUE TITLE
UriTemplate not evaluated correctly

### ISSUE THREAD

**Issue by it-ony** [danielgtaylor/aglio#138]
According to the specification https://github.com/apiaryio/api-blueprint/blob/master/API%20Blueprint%20Specification.md#form-style-query-variable an uri-template like /foo/bar{?a,b} should result in an uri /foo/bar?a=foo&b=bar if the parameters are set to `a=foo` and `b=bar`.

Right now the result looks /foo/bar{?a,b}.

**Comment by pksunkara** [comment]
@it-ony The URI will look `/foo/bar{?a,b}` in the documentation, but it will look as `/foo/bar?a=foo&b=bar` in the console since the parameters are specific to the console.

**Comment by danielgtaylor** [comment]
@it-ony the rendered documentation will show the URI template, not its evaluated result. Where are you expecting the evaluated result to be shown?

**Comment by it-ony** [comment]
@danielgtaylor I was expecting to show it where right now the template is shown, but in an evaluated format.

/foo/bar{?a,b} should just transformed to /foo/bar?a=fo0&b=bar. The values should be taken from the configured parameters, with their default values.

I tried the same api blueprint code at apiary.io and the display it in an evaluated format.

**Comment by pksunkara** [comment]
In Apiary, the evaluated result will be shown in the console only since parameters are specific to console. The non-evaluated is the one which will be shown in the documentation.

**Comment by it-ony** [comment]
Ok, what do you think about showing the defaults and also formatting the url? Query parameters shwosn as `{?a,b}` is not common for users reading the documentation. Using an uri template like `/foo/bar?a={a}&b={b}` will help out to show the uri in the documentation correctly, but seems wrong to me according to the specification.

**Comment by danielgtaylor** [comment]
@it-ony I like the idea of showing a generated example, but I worry about showing incorrect information. For example:

```apib
# My Resource [/frobs{?foo,bar}]
## Get a Frob [GET]
+ Parameters
  + foo: `fooString` - Passes a foo value. Mutually exclusive with `bar`!
  + bar: `barString` - Passes a bar value. Mutually exclusive with `foo`!
```

Now, you would generate `/frobs?foo=fooString&bar=barString` or something like that, which is **absolutely wrong** and will cause confusion.

**Comment by it-ony** [comment]
@danielgtaylor Is there a way to define dependencies with the apiblueprint's? As I guess the answer is no, I think the uri just gives you a hint of how to use it. If there are special rules, e.g. parameters cannot be used together such an uri, will help you more then a uri of the scheme /frobs{?foo,bar}.

As this is a GET request, I also ask myself, if there is any valid use-case for mutually exclusive parameter, or if just one parameter should win, if both are specified.

**Comment by danielgtaylor** [comment]
@it-ony I've been working on an implementation of this idea. I don't think the mutually exclusive options should be a blocker for now since this has the potential to help a lot of people with little or no URI template experience. Example of what I'm working on for the next release:

<img width="588" alt="screen shot 2015-10-28 at 17 14 33" src="https://cloud.githubusercontent.com/assets/106826/10806930/999211d0-7d97-11e5-9ce3-59270cfa8e65.png">

What do you think?

**Comment by it-ony** [comment]
Pretty cool!

**Comment by danielgtaylor** [comment]
This is now released in olio-theme-1.3.0. Please let me know if you find any problems with it!
